lib/experimental/fonts-library/class-wp-fonts-library.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+class WP_Fonts_Library {
+
+	const ALLOWED_FONT_MIME_TYPES = array(
+		'otf'   => 'font/otf',
+		'ttf'   => 'font/ttf',
+		'woff'  => 'font/woff',
+		'woff2' => 'font/woff2',
+	);
+
+	/**
+	 * Returns the absolute path to the fonts directory.
+	 *
+	 * @return string Path to the fonts directory.
+	 */
+	static public function get_fonts_directory() {
+		return path_join( WP_CONTENT_DIR, 'fonts' );
```

Comment:
This can be problematic perhaps. Seen quite a few cases in the past where the server is set to allow writes from PHP only to `wp-content/uploads`. Any particular reason why `wp-content/uploads/fonts` should not be used for fonts (assuming this is to store fonts uploaded by the users).

If `wp-content/uploads/fonts` is used, perhaps it can be excluded from the image library too. Thinking that'd be pretty easy.

Reply:
> This can be problematic perhaps. Seen quite a few cases in the past where the server is set to allow writes from PHP only to wp-content/uploads. Any particular reason why wp-content/uploads/fonts should not be used for fonts (assuming this is to store fonts uploaded by the users)?

Nope, any folder works for the purpose of storing the font assets. 

> Edit: actually another reason to store in the uploads directory is to perhaps allow users to manage local fonts through the (so called) Image Library, eventually some time in the future.

This PR is part of the Font Library. You can see screenscats of the font management functionality frontend here: https://github.com/WordPress/gutenberg/pull/50927 

> Alternatively the wp-content/fonts directory would have to be committed to git/svn, just like any other WP directory. Thinking that's actually better. In that case the create_fonts_directory() function would be pointless/shouldn't exist.

I'll try to change the implementation to leverage the existing upload functionality and the `uploads` folder.

Reply:
Thinking more about this, the `/fonts` directory will be part of WP, just like `/plugins`, `/themes`, `/uploads`, etc. right? Then it probably makes the most sense to just add it (in git/svn) under `/wp-content` like the other directories that hold various content/assets.

In that case the `define_fonts_directory()` and `create_fonts_directory()` would probably be better moved out of the class as they make sense only for the Gutenberg plugin. The code from them can perhaps be added to a "Gutenberg helper function" that runs on `plugins_loaded` and ensures the constant and the directory exist. Perhaps something like:
`
if ( ! defined( 'WP_FONTS_DIR' ) ) {
    define( 'WP_FONTS_DIR', WP_CONTENT_DIR . '/fonts';
}

if ( is_dir( WP_FONTS_DIR ) ) {
    wp_mkdir_p( WP_FONTS_DIR );
}
`

Reply:
I removed the constant and methods related to the WP_FONTS_DIR. With the [implementation of the `wp_hadle_upload` ](https://github.com/WordPress/gutenberg/pull/52704/commits/fd810862f8b09cf17f54a05a194ad4d2e9561021)we are using the wp-content/uploads/fonts path.